File: app/src/main/java/com/past/music/fragment/LocalMusicFragment.java
package com.past.music.fragment;


import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.past.music.adapter.MusicListAdapter;
import com.past.music.entity.MusicEntity;
import com.past.music.pastmusic.R;
import com.past.music.utils.MConstants;
import com.past.music.utils.MusicUtils;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/3/4 下午3:17
 * 描述：
 * 备注：Copyright © 2010-2017. gaojin All rights reserved.
 * =======================================================
 */

public class LocalMusicFragment extends BaseFragment {


    @BindView(R.id.local_music_recycle)
    RecyclerView mRecyclerView;

    private MusicListAdapter mAdapter;
    ArrayList<MusicEntity> musicList = new ArrayList<>();


    public static LocalMusicFragment newInstance(String param1, String param2) {
        LocalMusicFragment fragment = new LocalMusicFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    public LocalMusicFragment() {

    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_local_music, container, false);
        ButterKnife.bind(this, view);
        mAdapter = new MusicListAdapter(getContext());
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setAdapter(mAdapter);
        refreshAdapter();
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    public void refreshAdapter() {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(final Void... unused) {
                musicList.clear();
                musicList.addAll(MusicUtils.queryMusic(mContext, MConstants.START_FROM_LOCAL));
                mAdapter.updateDataSet(musicList);
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) {
                mAdapter.notifyDataSetChanged();
            }
        }.execute();
    }
}

File: app/src/main/java/com/se/music/fragment/LocalMusicFragment.kt
package com.se.music.fragment


import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.se.music.adapter.LocalFragmentAdapter
import com.se.music.pastmusic.R
import com.se.music.utils.MConstants
import com.se.music.utils.MusicUtils
import java.util.*

/**
 * A simple [Fragment] subclass.
 * Use the [KtUiLocalMusicFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class LocalMusicFragment : KtBaseFragment() {

    internal var mAdapter: LocalFragmentAdapter? = null
    private var mTabNames: List<String>? = null
    private var position: Int = 0
    private var mTabLayout: TabLayout? = null
    private var mViewPager: ViewPager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            position = arguments!!.getInt(POSITION)
        }
        val localCount = MusicUtils.queryMusic(context!!, MConstants.START_FROM_LOCAL).size
        val singerCount = MusicUtils.queryArtist(context!!).size
        val albumCount = MusicUtils.queryAlbums(context!!).size
        val folderCount = MusicUtils.queryFolder(context!!).size

        mTabNames = listOf(context!!.getString(R.string.local_music_song, localCount)
                , context!!.getString(R.string.local_music_singer, singerCount)
                , context!!.getString(R.string.local_music_album, albumCount)
                , context!!.getString(R.string.local_music_folder, folderCount))
    }

    override fun createContentView(inflater: LayoutInflater, container: ViewGroup?): View {
        val content = LayoutInflater.from(context).inflate(R.layout.fragment_local_music_kotlin, container, false)
        mTabLayout = content.findViewById(R.id.local_tab_layout)
        mViewPager = content.findViewById(R.id.local_view_pager)
        return content
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setTitle(context!!.getString(R.string.local_music_title))
        mAdapter = LocalFragmentAdapter(fm, mTabNames!!)
        mViewPager!!.adapter = mAdapter
        mViewPager!!.currentItem = position
        mTabLayout!!.setupWithViewPager(mViewPager)
        mAdapter!!.notifyDataSetChanged()
    }

    companion object {
        private val POSITION = "POSITION"

        fun newInstance(position: Int): LocalMusicFragment {
            val fragment = LocalMusicFragment()
            val args = Bundle()
            args.putInt(POSITION, position)
            fragment.arguments = args
            return fragment
        }
    }

}
